app/models/course/assessment/answer.rb

```diff
@@ -35,9 +35,13 @@ class Course::Assessment::Answer < ActiveRecord::Base
   def auto_grade!
     fail ArgumentError unless question.auto_gradable?
 
-    save!
-    create_auto_grading!
-    Course::Assessment::Answer::AutoGradingJob.new(auto_grading)
+    self.class.transaction do
+      save!
```

Comment:
Think you should do `raise ActiveRecord::Rollback unless save` ? otherwise the exception will be propagated